common/djangoapps/third_party_auth/utils.py

```diff
@@ -57,3 +76,19 @@ def validate_uuid4_string(uuid_string):
     except ValueError:
         return False
     return True
+
+
+def is_provider_saml(backend, kwargs):
+    """ Verify that the third party provider uses SAML """
+    current_provider = provider.Registry.get_from_pipeline({'backend': backend, 'kwargs': kwargs})
+    saml_providers_list = list(provider.Registry.get_enabled_by_backend_name('tpa-saml'))
+    return (current_provider and
+            current_provider.slug in [saml_provider.slug for saml_provider in saml_providers_list]), current_provider
+
+
+def is_enterprise_customer_user(provider_id, user):
+    """ Verify that the user linked to enterprise customer of current identity provider"""
+    enterprise_idp = EnterpriseCustomerIdentityProvider.objects.get(provider_id=provider_id)
+
+    return EnterpriseCustomerUser.objects.filter(enterprise_customer=enterprise_idp.enterprise_customer,
```

Comment:
What if no enterprise_idp is found?

Reply:
That is not possible as the user will be coming from an IDP and we redirect the user to IDP only if the provider id is found. Here is the reference code where we redirect the user.
https://github.com/edx/edx-platform/blob/aed3d8cbfcd81ab4320224743eeff0115f953d24/openedx/core/djangoapps/user_authn/views/login_form.py#L170

Reply:
it seems to me that we're putting a lot of faith in the workflow by assuming that this code only gets executed in that context.  I think it would be worthwhile to handle for a null `enterprise_idp` condition and include logging, accordingly.

Reply:
We have also put this code in a try block and catching and logging the exception if something like this will happen. I have tested this locally and it will log the exception `EnterpriseCustomerIdentityProvider matching query does not exist.:` with all the other details like use email, provider id, and user id.